crates/pathfinder/src/storage/merkle_tree.rs

```diff
@@ -334,6 +338,42 @@ impl<'tx, 'queries> RcNodeStorage<'tx, 'queries> {
             0
         })?;
 
+        let count = match count {
+            0 => {
+                let existing = self
+                    .get(key)
+                    .context("Reading existing node")?
+                    .context("Node should exist since insert failed")?;
+
+                match &existing {
+                    PersistedNode::Leaf => {
+                        // We no longer store leaf nodes, but it is possible for them to still exist in some databases.
+                        // It is therefore okay to overwrite them.
+                        let new_count = self
+                            .transaction
+                            .execute(
+                                &self.queries.upsert,
```

Comment:
Is upsert needed elsewhere? This is now an update basically, since we know the row exists?

Reply:
Ah yeah `update` makes more sense.